### PR TITLE
fixes for piral 0.15

### PR DESCRIPTION
**Pull request by luckyluggi**
# New Pull Request

For more information, see the `CONTRIBUTING` guide.

## Prerequisites

Please make sure you can check the following boxes:

- [ ] I have read the **CONTRIBUTING** document
- [ ] My code follows the code style of this project
- [ ] All new and existing tests passed

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes

### Description
This makes piral blazor compatible with piral 0.15

### Remarks

[Optionally place any follow-up comments, remarks, observations, or notes here for future reference]

### CODE REVIEW

**Comment by luckyluggi** [comment]
@FlorianRappl  With these changes the blazor pilets are building again, but the web page stays blank.
`<div id="app"></div>` is not replaced with the content.
Where could this be fixed?

**Comment by FlorianRappl** [comment]
That rather sounds like an issue in the emulator. The emulator renders, so if it does not then something is wrong with the emulator.

**Comment by luckyluggi** [comment]
ah yes you're right. the app shell does currently not work. i tried to create an mvp, but as soon as i install piral-blazor@latest & blazor@latest into the blank app, shell it stops working.
(It looks like piral-core gets downgraded to 0.14 again and it also cant find react.)

So i guess the blazor npm packages might need an update to work with 0.15?

But i think this PR is probablyfine as it is now.

**Comment by FlorianRappl** [comment]
Yeah the PR is fine as it is (except for the codegen modifications - these need to be in the respective source files, otherwise they'd be overwritten in the future). The `blazor` npm package has no dependency to React or Piral - so it should be independent of that.

File: src/Piral.Blazor.Tools/content/src/blazor.codegen [hunk 1]
```diff
@@ -293,16 +294,33 @@ function toFramework(files) {
...
+  const appdir = findAppDir(piralPiletFolder, instanceName);
   const manifestSource = staticAssets.Assets.find((m) => m.AssetTraitName === "BlazorWebAssemblyResource" && m.AssetTraitValue === "manifest" && m.RelativePath.endsWith(bbjson));
   if (!manifestSource) {
     throw new Error(`Could not find the "${bbjson}" in ${swajson}. Something seems to be wrong.`);
   }
   const appFrameworkDir = (0, import_path5.resolve)(appdir, "app", "_framework");
   const bbAppShellPath = (0, import_path5.resolve)(appFrameworkDir, bbjson);
   const blazorInAppshell = (0, import_fs4.existsSync)(bbAppShellPath);
-  const shellPackagePath = (0, import_path5.resolve)(appdir, pjson);
+  const shellPackagePath = (0, import_path5.resolve)(appdir, packageJsonFilename);
```
Comment (FlorianRappl) [inline src/Piral.Blazor.Tools/content/src/blazor.codegen:323]: Danger - the whole file is generated.

If you want to change the contents of the file you should change the sources: https://github.com/smapiot/Piral.Blazor/tree/blazor-6.0/src/blazor-codegen/src

For instance, the `prepare` is in https://github.com/smapiot/Piral.Blazor/blob/blazor-6.0/src/blazor-codegen/src/prepare.ts

**Review by FlorianRappl — APPROVED** [review]

**Pull request merged**